src/server/game/Warden/WardenWin.cpp

```diff
@@ -413,26 +421,11 @@ void WardenWin::HandleCheckResult(ByteBuffer &buff)
                 TC_LOG_DEBUG("warden", "RESULT %s passed CheckId %u account Id %u", EnumUtils::ToConstant(check.Type), id, _session->GetAccountId());
                 break;
             }
-            case LUA_STR_CHECK:
+            case LUA_EVAL_CHECK:
             {
-                uint8 Lua_Result;
-                buff >> Lua_Result;
+                buff.read<uint8>();
```

Comment:
you need to read_skip the rest if != 0

Reply:
No, in fact current Warden is broken, and adding a `LUA_STR_CHECK` to the DB in current Warden will corrupt data after it.

Result should only be read if `Lua_Result` is zero, not nonzero.

Reply:
i mean yeah i know the check is broken as far as its implementation in module is concerned but then you still have data if = 0

Reply:
I have only ever observed result 1 for `::Execute` substitution, I believe it to be the return value from one of the lua library functions.

Anyway, added `read_skip`.